### PR TITLE
Add missing fields to file and folder models

### LINKED ISSUE #581: Missing fields in BoxFile & BoxFolder
- [x] I have checked that the [SDK documentation][sdk-docs] and [API documentation][api-docs] doesn't solve my issue

### Description of the Issue
The `BoxFile` class doesn't expose fields that exist on the File object.

For example, `is_externally_owned`. which exists in the general API reference: https://developer.box.com/reference#file-object
And also in other client SDKs: https://github.com/box/box-java-sdk/blob/master/src/main/java/com/box/sdk/BoxFile.java#L1723

Missing fields for `BoxFile`:
- expires_at
- allowed_invitee_roles
- has_collaborations
- is_externally_owned

Missing fields for `BoxFolder`:
- expires_at
- is_collaboration_restricted_to_enterprise

### Versions Used
.NET SDK: 3.18.0
.NET Framework/Core: .NET Core 3 preview 7

### PR DESCRIPTION
**Pull request by mattwiller**
Fixes #581

### CODE REVIEW

**Comment by boxcla** [comment]
Verified that @mattwiller has signed the CLA. Thanks for the pull request!

File: Box.V2/Models/BoxFolder.cs [hunk 1]
```diff
@@ -113,5 +115,17 @@ public class BoxFolder : BoxItem
...
+        /// The date when the folder will be automatically deleted due to item expiration settings.
+        /// </summary>
+        [JsonProperty(PropertyName = FieldExpiresAt)]
+        public DateTime? ExpiresAt { get; protected set; }
+
+        /// <summary>
+        /// The date when the folder will be automatically deleted due to item expiration settings.
+        /// </summary>
+        [JsonProperty(PropertyName = FieldIsCollaborationRestrictedToEnterprise)]
+        public bool? IsCollaborationRestrictedToEnterprise { get; protected set; }
```
Comment (carycheng) [inline Box.V2/Models/BoxFolder.cs]: Should this be `Boolean?`

File: Box.V2/Models/BoxFile.cs [hunk 2]
```diff
@@ -113,5 +118,29 @@ public class BoxFile : BoxItem
...
+        /// Whether the item has collaborations or not.
+        /// </summary>
+        [JsonProperty(PropertyName = FieldHasCollaborations)]
+        public bool? HasCollaborations { get; protected set; }
+
+        /// <summary>
+        /// Whether the item is owned by an entity external to the user's enterprise.
+        /// </summary>
+        [JsonProperty(PropertyName = FieldIsExternallyOwned)]
+        public bool? IsExternallyOwned { get; protected set; }
```
Comment (carycheng) [inline Box.V2/Models/BoxFile.cs]: Should this be `Boolean?`

File: Box.V2.Test/BoxFilesManagerTest.cs [hunk 3]
```diff
@@ -81,7 +82,85 @@ public async Task GetCollaborationsCollectionAsync_ValidResponse_NextMarker()
...
+            string responseString = @"{
```
Comment (carycheng) [inline Box.V2.Test/BoxFilesManagerTest.cs]: Big spacing after `@"{` I'm not sure if this matters or if it's just github spacing?

File: Box.V2.Test/BoxFoldersManagerTest.cs [hunk 4]
```diff
@@ -329,7 +329,90 @@ public async Task GetFolderInformation_ValidResponse_ValidFolder()
...
+                    ContentString = @"{
```
Comment (carycheng) [inline Box.V2.Test/BoxFoldersManagerTest.cs]: Spacing here as well?

File: Box.V2/Models/BoxFolder.cs (hunk 1 above)
Comment (mattwiller) [reply Box.V2/Models/BoxFolder.cs]: In C#, `bool` is just an alias for `Boolean` — `bool?` is correct and allows a `null` value.

File: Box.V2/Models/BoxFile.cs (hunk 2 above)
Comment (mattwiller) [reply Box.V2/Models/BoxFile.cs]: In C#, `bool` is just an alias for `Boolean` — `bool?` is correct and allows a `null` value.

File: Box.V2.Test/BoxFilesManagerTest.cs (hunk 3 above)
Comment (mattwiller) [reply Box.V2.Test/BoxFilesManagerTest.cs]: It's just GitHub; it's showing a diff between one really long line that wraps several times and a much shorter line (where all the other content has been broken onto subsequent lines)

**Pull request merged**